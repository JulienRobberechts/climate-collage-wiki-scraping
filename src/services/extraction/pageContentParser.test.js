const { getContent } = require('../fileServices/readFile');
const { parsePageContent } = require('./pageContentParser');

const cardContent1FilePath = './src/tests/data/carte1.html';

describe('parsePageContent', () => {
  it('read', async () => {
    const content = await getContent(cardContent1FilePath);
    expect(content).toMatch(/modifier le wikicode/gm);
  });
  it('parse', async () => {
    const content = await getContent(cardContent1FilePath);
    const result = parsePageContent(content);
    expect(result).toStrictEqual({
      causes: [
      ],
      consequences: [
        "/wiki/index.php?title=Fr-fr_adulte_carte_2_industrie",
        "/wiki/index.php?title=Fr-fr_adulte_carte_3_b%C3%A2timent",
        "/wiki/index.php?title=Fr-fr_adulte_carte_4_transport",
        "/wiki/index.php?title=Fr-fr_adulte_carte_8_agriculture",
      ],
    });
  });
});
