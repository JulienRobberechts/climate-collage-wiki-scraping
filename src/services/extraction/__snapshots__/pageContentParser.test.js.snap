// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse Page Content parse BackDescription card 3 1`] = `
"\\\\nL\\\\u2019utilisation du b\\\\u00e2timent (logement et services) utilise des \\\\u00e9nergies fossiles et de
      l'\\\\u00e9lectricit\\\\u00e9.\\\\nCela repr\\\\u00e9sente 20% des Gaz \\\\u00e0 Effet de Serre (GES).\\\\n"
`;

exports[`parse Page Content parse Explanation card 1 1`] = `
"C'est par l\\\\u00e0 que tout commence... C'est aussi par
    l\\\\u00e0 que commence le 5\\\\u00e8me rapport du GIEC, dans l'avant-propos de son r\\\\u00e9sum\\\\u00e9, il est mis en
    \\\\u00e9vidence que la science montre \\\\u00e0 pr\\\\u00e9sent avec 95Â % de certitude que depuis le milieu du XXe
      si\\\\u00e8cle, l\\\\u2019activit\\\\u00e9 humaine est la cause principale du r\\\\u00e9chauffement observ\\\\u00e9.

    Cette carte peut \\\\u00eatre consid\\\\u00e9r\\\\u00e9e...\\\\nsoit comme la cause de l'ensemble des cartes secteurs \\\\u00e9conomiques (Industrie, Utilisation des b\\\\u00e2timents, Transport, Agriculture),\\\\nsoit comme un titre de l'ensemble de ces cartes
      (qui sont alors regroup\\\\u00e9es dans une grosse \\\\u201cpatate\\\\u201c)."
`;
